packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -240,6 +294,35 @@ class ToOOXMLVisitor {
 
                             this.globalOOXML += ooxml;
                             this.tags = [];
+                        } else if (this.getClass(subNode) === TRANSFORMED_NODES.optional) {
+                            if (!subNode.hasSome) {
+                                this.isTraversingFalseCondition = true;
+                            }
+                            // traverse the whenSome properties now
+                            this.traverseNodes(subNode.whenSome, properties, true);
+
+                            if (!subNode.hasSome) {
+                                this.isTraversingFalseCondition = false;
+                            } else {
+                                this.isTraversingFalseCondition = true;
+                            }
+                            this.isTraversingElseCondition = true;
+                            // traverse whenNone properties now
+                            this.traverseNodes(subNode.whenNone, properties, true);
+                            let ooxml = '';
+                            for (let tag of this.conditionalOrOptionalTags) {
+                                ooxml += tag;
+                            }
+                            this.isTraversingFalseCondition = false;
+                            this.isTraversingElseCondition = false;
+                            this.conditionalOrOptionalTags = [];
+
+                            const tag = subNode.name;
+                            const type = subNode.elementType;
+                            this.createOrUpdateCounter(tag, type);
+                            const title = `${tag.toUpperCase()[0]}${tag.substring(1)}${this.counter[tag].count}`;
+                            let optionalTag = OPTIONAL_RULE(title, tag, ooxml, type);
+                            this.tags = [...this.tags, optionalTag];
```

Comment:
It's so involving that I think you probably need to abstract it into a different function after considering my strongly suggested recommendation, written above, for refactoring.

Also, the way you have assigned the boolean values are redundant.
**BAD**
```js
if (!subNode.hasSome) {
    this.isTraversingFalseCondition = false;
} else {
    this.isTraversingFalseCondition = true;
}
```
**OK**
```js
this.isTraversingFalseCondition = subNode.hasSome;
```